OCPBUGS#59417 GCP Cloud Organization Policy and VPC-SC Requirements for OpenShift Installer (#102568)
openshift/openshift-docs: modules/installation-gcp-organization-policies.adoc

* VPC Service Controls egress rules

.Prerequisites

* You created a project to host your cluster.

.Procedure

. Modify the `constraints.compute.trustedImageProjects` constraint in your {gcp-short} project and add `projects/rhcos-cloud` to the `allowedValues` list, as in the following example. Alternatively, you can delete the `constraints.compute.trustedImageProjects` constraint.
+
[source,yaml]
----
constraint: constraints/compute.trustedImageProjects
listPolicy:
 allowedValues:
    - projects/rhcos-cloud
----
. Modify the `iam.allowedPolicyMemberDomains` constraint to allow the service account that the installation program uses to authenticate with {gcp-short} and create storage.
. Disable the `storage.publicAccessPrevention` constraint so that the installation program can access your project's cloud storage.
. Create an egress rule that allows access to the {op-system} project, as in the following example:
+
[source,yaml]
----
- egressTo:
    roles:
    - <role_name>
    resources:
    - projects/902460926346
  egressFrom:
    identityType: <service_account>>
----
where:
+
<role_name>:: Specifies the IAM role that you created for the installation program.

> <service_account>:: Specifies the name of the installation program service account.

Reviewer: @lnguyen1401 Do you know which role or permissions are required? 

@bscott-rh Per Google [doc](https://docs.cloud.google.com/vpc-service-controls/docs/ingress-egress-rules#egress-rules-reference), "egressFrom.identityType" should be ANY_IDENTITY, ANY_USER_ACCOUNT, or ANY_SERVICE_ACCOUNT instead. And "egressFrom.identities" can be used to list the service accounts and etc.

// Module included in the following assemblies:
//
// * installing/installing_gcp/installing-gcp-account.adoc

:_mod-docs-content-type: PROCEDURE
[id="installation-gcp-organization-policies_{context}"]
= Configuring {gcp-short} organization policies and VPC service controls

[role="_abstract"]
{product-title} requires that you modify or remove some organization policies and VPC service controls. The following policies must be modified or removed:

* `compute.trustedImageProjects`
* `iam.allowedPolicyMemberDomains`
* `storage.publicAccessPrevention`
* `compute.requireOsLogin`
* VPC Service Controls egress rules

.Prerequisites

* You created a project to host your cluster.

.Procedure

> . Modify the `constraints.compute.trustedImageProjects` constraint in your {gcp-short} project and add `projects/rhcos-cloud` to the `allowedValues` list, as in the following example:

Reviewer 2: Can we add 'Or remove the organization policy if applicable' ?

+
[source,yaml]
----
constraint: constraints/compute.trustedImageProjects
listPolicy:
 allowedValues:
    - projects/rhcos-cloud
----
. Modify the `iam.allowedPolicyMemberDomains` constraint to allow the service account that the installation program uses to authenticate with {gcp-short} and create storage.
. Disable the `storage.publicAccessPrevention` constraint so that the installation program can access your project's cloud storage.

> . If you intend to access nodes using SSH, either disable `compute.requireOsLogin` at the organizational level, or configure the credentials provided by the installation program as an exception to the OS Login limitations.

Reviewer 2: Please disregard the comment above and le'ts remove the entry on comput.requireOsLogin

> . If you intend to access nodes using SSH, either disable `compute.requireOsLogin` at the organizational level, or configure the credentials provided by the installation program as an exception to the OS Login limitations.

Author: @lnguyen1401 Thanks Linh, I've updated the PR with your feedback

// Module included in the following assemblies:
//
// * installing/installing_gcp/installing-gcp-account.adoc

:_mod-docs-content-type: PROCEDURE
[id="installation-gcp-organization-policies_{context}"]
= Configuring {gcp-short} organization policies and VPC service controls

[role="_abstract"]

> {product-title} requires that you modify or remove some organization policies and VPC service controls when installing a cluster on {gcp-first}. If your organization uses the following policies, they must be modified or removed:

Reviewer 3: ```suggestion
{product-title} requires that you modify or remove some organization policies and VPC service controls when installing a cluster on {gcp-first}. 

If your organization uses the following policies, you must modify or remove them:
```
the intro to a list cannot be part of an abstract; also passive-to-active voice

* `compute.trustedImageProjects`
* `iam.allowedPolicyMemberDomains`
* `storage.publicAccessPrevention`
* VPC Service Controls egress rules

.Prerequisites

* You created a project to host your cluster.

.Procedure

> . Modify the `constraints/compute.trustedImageProjects` constraint in your {gcp-short} project and add `projects/rhcos-cloud` to the `allowedValues` list, as in the following example. Alternatively, you can delete the `constraints/compute.trustedImageProjects` constraint.

Reviewer 3: ```suggestion
. Modify the `constraints/compute.trustedImageProjects` constraint in your {gcp-short} project and add `projects/rhcos-cloud` to the `allowedValues` list, as in the following example:
```

+
[source,yaml]
----
constraint: constraints/compute.trustedImageProjects
listPolicy:
 allowedValues:
    - projects/rhcos-cloud

> ----

Reviewer 3: ```suggestion
----
+
Alternatively, you can delete the `constraints/compute.trustedImageProjects` constraint.
```

> . Modify the `iam.managed.allowedPolicyMembers` constraint to allow the service account that the installation program uses to authenticate with {gcp-short} and create storage as in the following example:

Reviewer 3: ```suggestion

. Modify the `iam.managed.allowedPolicyMembers` constraint to allow the service account that the installation program uses to authenticate with {gcp-short} and create storage as shown in the following example:
```
DITA might delete this next step without the line space

+
[source,yaml]
----
name: organizations/<organization_id>/policies/iam.managed.allowedPolicyMembers
spec:
rules:
 - enforce: true
   parameters:
     allowedMemberSubjects:
       - <allowed_member>
----

> where:

Reviewer 3: ```suggestion
+
where:
```

+
`<organization_id>`:: Specifies the numeric ID of your {gcp-short} organization.
`<allowed_member>`:: Specifies the IAM role that you created for the installation program. You can specify a service account in the format of "serviceAccount:example-service-account@example.com" or a user in the format of "user:example-user@example.com".

> . Disable the `storage.publicAccessPrevention` constraint so that the installation program can access your project's cloud storage.

Reviewer 3: ```suggestion
. Disable the `storage.publicAccessPrevention` constraint so that the installation program can access your the cloud storage for your project.
```
no possessives

> . Create an egress rule that allows access to the {op-system} project, as in the following example:

Reviewer 3: ```suggestion

. Create an egress rule that allows access to the {op-system} project, as in the following example:
```

+
[source,yaml]
----
- egressFrom:
    identityType: ANY_IDENTITY
    identities:
      - "serviceAccount:<installation_program_sa>@<project_id>.iam.gserviceaccount.com"
      - "user:<admin_email>"
  egressTo:
    resources:
      - <project_id>
    operations:
      - serviceName: "compute.googleapis.com"
        methodSelectors: [{method: "*"}]
      - serviceName: "storage.googleapis.com"
        methodSelectors: [{method: "*"}]
      - serviceName: "dns.googleapis.com"
        methodSelectors: [{method: "*"}]
      - serviceName: "iam.googleapis.com"
        methodSelectors: [{method: "*"}]
      - serviceName: "cloudresourcemanager.googleapis.com"
        methodSelectors: [{method: "*"}]
      - serviceName: "serviceusage.googleapis.com"
        methodSelectors: [{method: "*"}]
      - serviceName: "artifactregistry.googleapis.com"
        methodSelectors: [{method: "*"}]
      - serviceName: "deploymentmanager.googleapis.com"
        methodSelectors: [{method: "*"}]
----

> where:

Reviewer 3: ```suggestion
+
where:
```

Reviewer 3: DITA will strip content unless it is attached to the step, see [this example](https://github.com/openshift/openshift-docs/pull/102276/changes#diff-a7b94c9c8cd49537db0616cf37189d6e9a21954425846b4ed3e540b36a6d1d8eR58).
